### PR TITLE
Update the display of lesson name and number to be based on lockable and has_lesson_plan

### PR DESCRIPTION
**Pull request by dmcavoy**
## Background

Previous PRs have:
- Added has_lesson_plan to lesson model
- Set has_lesson_plan for existing lessons based on lesson lockable and script has_lesson_plan setting
- Updated relative position to be based on lockable and has_lesson_plan

## This PR

Updates the display of lessons so that a lesson that is numbered will show that number in different places on the front end. See all the examples below. I reviewed the progress changes with Amanda.

#### Lesson Navigation Dropdown

Don't show lessons without lesson plans in the navigation list.

Screen Shot 2021-01-28 at 7 33 05 AM

#### Script Overview: Lesson (Lockable and Has Lesson Plan)

Screen Shot 2021-01-27 at 4 18 43 PM

#### Script Overview: Survey (Lockable and does not have lesson plan)

Screen Shot 2021-01-27 at 4 19 16 PM

#### Progress View Lesson Selector

Note: I tested this locally by updating one of the CSP 20-21 lessons because CSP 21-22 is not assignable yet. You will see that it says 14: Lesson 14. That is because "Lesson 14" is part of the name of the lesson. When I update those lessons officially that name will get updated but didn't do it hear for time.

Note 2: This and the other progress view examples below show the change for a lesson that is lockable and has a lesson plan.

Before: 
Screen Shot 2021-01-27 at 4 27 39 PM

After:
Screen Shot 2021-01-27 at 4 25 58 PM

#### Progress Summary View

Before: 
Screen Shot 2021-01-27 at 4 27 33 PM

After:
Screen Shot 2021-01-27 at 4 25 30 PM

#### Progress Detail View

Before: (Look at the last lesson)
Screen Shot 2021-01-27 at 4 27 46 PM

After:
Screen Shot 2021-01-28 at 7 59 48 PM

## Links

- [spec](https://docs.google.com/document/d/1sAAVnwv3rXQEK-qa8JnkNXFfZ0vjWQyzJ9Yx_pG4bdg/edit#)
- [PLAT-534](https://codedotorg.atlassian.net/browse/PLAT-534)
- [PLAT-723](https://codedotorg.atlassian.net/browse/PLAT-723)

## Testing story

- Manually tested the different pages (see screen shots)
- Updated the test for processedStages to check that it works correctly for numbering the correct lessons
- Checked that you can summarize a lesson with and without lesson plan correctly
- Update test to make sure summarize_for_lesson_show in script filters to only lessons with lesson plans
- Updated a couple other tests as well

### CODE REVIEW

**Review by bethanyaconnor — APPROVED** [review]
Looks good to me!

File: dashboard/test/factories/factories.rb [hunk 1]
```diff
@@ -790,7 +790,7 @@
     end
 
     # relative_position is actually the same as absolute_position in our factory
-    # (i.e. it doesnt try to count lockable/non-lockable)
+    # (i.e. it doesnt try to count lockable lesson without lesson plan/all other lessons)
```
Comment (davidsbailey) [inline dashboard/test/factories/factories.rb]: I see what you did here, but the only way I can make sense of the new comment is by looking at the old comment to see what you did. here's one option:
```
    # i.e. it doesn't try to count lockable lessons without lesson plans separately from all other lessons, which is what we normally do for relative position.
```

File: apps/src/code-studio/progressRedux.js [hunk 2]
```diff
@@ -298,21 +298,21 @@ export const getLevelResult = level => {
 };
 
 /**
- * Does some processing of our passed in stages, namely
+ * Does some processing of our passed in lesson, namely
  * - Removes 'hidden' field
- * - Adds 'stageNumber' field for non-lockable, non-PLC stages
+ * - Adds 'stageNumber' field for non-lockable lessons or lessons with lesson plans, non-PLC lessons
```
Comment (davidsbailey) [inline apps/src/code-studio/progressRedux.js]: does this mean (lessons or lessons with lesson plans) which are also non-PLC ? can you clarify?

**Review by davidsbailey — COMMENTED** [review]
Code changes look good. How are you feeling about the chances of this causing a regression?

**Comment by dmcavoy** [comment]
@davidsbailey I feel fairly confident that this is a safe change to make

**Review by davidsbailey — APPROVED** [review]
This looks super thorough, Dani! Great work!

**Pull request merged**